jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6134 @@
+	 * @param destObj object to copy into
+	 * @param destOffset location in destObj to start copy
+	 * @param size the number of bytes to be copied
+	 * 
+	 * @throws IllegalArgumentException if srcOffset is illegal in srcObj, 
+	 * if destOffset is illegal in destObj, or if size is invalid
+	 */
+	public void copyMemory(Object srcObj, long srcOffset, Object destObj, long destOffset, long size) {
+		copyMemoryChecks(srcObj, srcOffset, destObj, destOffset, size);
+
+		if (0 != size) {
+			copyMemory0(srcObj, srcOffset, destObj, destOffset, size);
+		}
+	}
+
+	/**
+	 * Copy specified bytes from source address to destination
+	 * 
+	 * @param srcAddress address to start copy
+	 * @param destAddress address to start copy
+	 * @param size the number of bytes to be copied
+	 * 
+	 * @throws IllegalArgumentException if srcAddress or destAddress
+	 *  is illegal, or if size is invalid
+	 */
+	public void copyMemory(long srcAddress, long destAddress, long size) {
+		copyMemory(null, srcAddress, null, destAddress, size);
+	}
+
+	/**
+	 * Swap specified bytes between source object and destination
+	 * 
+	 * @param srcObj object to copy from 
+	 * @param srcOffset location in srcObj to start swap
+	 * @param destObj object to copy into
+	 * @param destOffset location in destObj to start swap
+	 * @param srcSize the number of bytes to be swapped in source
+	 * @param destSize the number of bytes to be swapped in destination
+	 * 
+	 * @throws IllegalArgumentException if srcOffset is illegal in srcObj, 
+	 * if destOffset is illegal in destObj, or if sizes are invalid
+	 */
+	public void copySwapMemory(Object srcObj, long srcOffset, Object destObj, long destOffset, long srcSize,
+			long destSize) {
+		copySwapMemoryChecks(srcObj, srcOffset, destObj, destOffset, srcSize, destSize);
+
+		if (0 != srcSize) {
+			copySwapMemory0(srcObj, srcOffset, destObj, destOffset, srcSize, destSize);
+		}
+	}
+
+	/**
+	 * Swap specified bytes between source address and destination
+	 * 
+	 * @param srcAddress location to start swap
+	 * @param destAddress location to start swap
+	 * @param srcSize the number of bytes to be swapped at source
+	 * @param destSize the number of bytes to be swapped at destination
+	 * 
+	 * @throws IllegalArgumentException if srcAddress or destAddress is illegal, 
+	 *  or if sizes are invalid
+	 */
+	public void copySwapMemory(long srcAddress, long destAddress, long srcSize, long destSize) {
+		copySwapMemory(null, srcAddress, null, destAddress, srcSize, destSize);
+	}
+
+	/**
+	 * Removes the block from the list. Note that the pointers are located 
+	 * immediately before the startIndex value.
+	 * 
+	 * @param startIndex memory address
+	 * 
+	 * @throws IllegalArgumentException if startIndex is not valid
+	 */
+	public void freeMemory(long startIndex) {
+		freeMemoryChecks(startIndex);
+
+		if (0 != startIndex) {
+			freeMemory0(startIndex);
+		}
+	}
+
+	/**
+	 * Returns byte offset to start of class or interface
```

Comment:
Is the class or interface part correct? I thought its the offset to where the field is stored in the object.